### PR TITLE
Fix bufr sounding job

### LINKED ISSUE #1832: Bufr sounding job broken after logf file renamed
### What is wrong?

Error message in log:
```
+ exgfs_atmos_postsnd.sh[74]: [[ ! -f /lfs/h2/emc/ptmp/kate.friedman/comrot/test96/gfs.20211221/00//model_data/atmos/history/gfs.t00z.logf012.txt ]]
+ exgfs_atmos_postsnd.sh[75]: sleep 10
++ exgfs_atmos_postsnd.sh[76]: expr 359 + 1
+ exgfs_atmos_postsnd.sh[76]: ic=360
+ exgfs_atmos_postsnd.sh[81]: '[' 360 -ge 360 ']'
+ exgfs_atmos_postsnd.sh[83]: err_exit 'COULD NOT LOCATE logf012 file AFTER 1 HOUR'
```

### What should have happened?

The RUN.tHHz.atm.logfFFF.txt file should have been seen/used but the scripts still look for the file without the ".atm" part of the name.

### What machines are impacted?

All or N/A

### Steps to reproduce

Run the gfspostsnd job on any platform as part of a gfs suite.

### Additional information

The logf file was renamed recently but the bufr sounding job scrips were not updated to reflect the change.

### Do you have a proposed solution?

Update the logf filename in the relevant bufr sounding scripts.

### PR DESCRIPTION
**Pull request by KateFriedman-NOAA**
# Description

This PR gets the bufr sounding job working again by making two changes:

1. updating the logf filename within the relevant scripts to match the new naming convention that includes ".atm"
2. updating to a newer gfs-utils repo hash that includes a change to file length in the gfs_bufr code (resolves failure in testing), see https://github.com/NOAA-EMC/gfs-utils/issues/21 & https://github.com/NOAA-EMC/gfs-utils/pull/22

Resolves #1832
Refs https://github.com/NOAA-EMC/gfs-utils/issues/21

# Type of change

- Bug fix (fixes something broken)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?

- Low res cycled test on WCOSS2 with `DO_BUFRSND="YES"` set and the gfspostsnd job included in the cycle.
- CI tests on Hera and Orion.

# Checklist
- [ ] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by KateFriedman-NOAA** [comment]
Error in log (PR/1834/RUNTESTS/COMROT/C48_ATM_5a4e4de6/logs/2021032312/gfspostsnd.log) on Hera:

```
+ gfs_bufr.sh[83]: srun -l --export=ALL -n 40 /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1834/global-workflow/exec/gfs_bufr.x
 0: forrtl: severe (66): output statement overflows record, unit -5, file Internal Formatted Write
 0: Image              PC                Routine            Line        Source
 0: gfs_bufr.x         000000000094B4AE  Unknown               Unknown  Unknown
 0: gfs_bufr.x         0000000000994EF1  Unknown               Unknown  Unknown
 0: gfs_bufr.x         0000000000406DF2  buff_                      31  buff.f
 0: gfs_bufr.x         0000000000408F35  MAIN__                    270  gfsbufr.f
 0: gfs_bufr.x         000000000040615E  Unknown               Unknown  Unknown
 0: libc-2.17.so       00002B80B6A18555  __libc_start_main     Unknown  Unknown
 0: gfs_bufr.x         0000000000406069  Unknown               Unknown  Unknown
srun: error: h3c39: task 0: Exited with exit code 66
srun: launch/slurm: _step_signal: Terminating StepId=48994302.9
+ gfs_bufr.sh[1]: postamble gfs_bufr.sh 1693581047 66
```

May need to run additional tests on Hera to investigate.

The issue with the log filename was at least resolved:

```
+ exgfs_atmos_postsnd.sh[74]: [[ ! -f /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1834/RUNTESTS/COMROT/C48_ATM_5a4e4de6/gfs.20210323/12//model_data/atmos/history/gfs.t12z.atm.logf012.txt ]]
+ exgfs_atmos_postsnd.sh[78]: break
+ exgfs_atmos_postsnd.sh[88]: '[' 012 -gt 0 ']'
+ exgfs_atmos_postsnd.sh[89]: export FINT=3
+ exgfs_atmos_postsnd.sh[89]: FINT=3
+ exgfs_atmos_postsnd.sh[92]: /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1834/global-workflow/ush/gfs_bufr.sh
Begin gfs_bufr.sh at Fri Sep  1 15:10:28 UTC 2023
```

**Comment by KateFriedman-NOAA** [comment]
> Error in log (PR/1834/RUNTESTS/COMROT/C48_ATM_5a4e4de6/logs/2021032312/gfspostsnd.log) on Hera:
> 
> ```
> + gfs_bufr.sh[83]: srun -l --export=ALL -n 40 /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1834/global-workflow/exec/gfs_bufr.x
>  0: forrtl: severe (66): output statement overflows record, unit -5, file Internal Formatted Write
>  0: Image              PC                Routine            Line        Source
>  0: gfs_bufr.x         000000000094B4AE  Unknown               Unknown  Unknown
>  0: gfs_bufr.x         0000000000994EF1  Unknown               Unknown  Unknown
>  0: gfs_bufr.x         0000000000406DF2  buff_                      31  buff.f
>  0: gfs_bufr.x         0000000000408F35  MAIN__                    270  gfsbufr.f
>  0: gfs_bufr.x         000000000040615E  Unknown               Unknown  Unknown
>  0: libc-2.17.so       00002B80B6A18555  __libc_start_main     Unknown  Unknown
>  0: gfs_bufr.x         0000000000406069  Unknown               Unknown  Unknown
> srun: error: h3c39: task 0: Exited with exit code 66
> srun: launch/slurm: _step_signal: Terminating StepId=48994302.9
> + gfs_bufr.sh[1]: postamble gfs_bufr.sh 1693581047 66
> ```

@BoCui-NOAA would you be able to take a look at the gfs_bufr code error that happened in a small CI test on Hera? See error above and saved test here on Hera: 

/scratch1/NCEPDEV/global/Kate.Friedman/expdir/CI_TEST_LOGS
see bottom of log: /scratch1/NCEPDEV/global/Kate.Friedman/expdir/CI_TEST_LOGS/COMROT/C48_ATM_5a4e4de6/logs/2021032312/gfspostsnd.log

The gfspostsnd (bufr sounding) job worked fine in other CI tests on Hera and Orion (including the same test configuration on Orion). Also, ran ok in a short C96C48 ATM-only cycled test on WCOSS2. Output from WCOSS2 test is here on Dogwood (ran on Cactus, copied logs to Dogwood): 

/lfs/h2/emc/global/noscrub/kate.friedman/expdir/test96/
logs: /lfs/h2/emc/global/noscrub/kate.friedman/expdir/test96/COMROT_LOGS/2021122*/*gfspostsnd*

Thanks for any help you can provide for this code!

**Comment by BoCui-NOAA** [comment]
@KateFriedman-NOAA I will take a look the codes and test log files.

**Comment by BoCui-NOAA** [comment]
@KateFriedman-NOAA  I checked code buff.f line 31

write(fnbufr,fmto) dird(1:lss),istat(np),jdate

This is a write statement. fnbufr has a length of 150 characters. The dird is the output path. In the failed test, 

DIRD    = /scratch1/NCEPDEV/global/Terry.McGuinness/GFS_CI_ROOT/PR/1834/RUNTESTS/COMROT/C48_ATM_5a4e4de6/gfs.20210323/12//products/a

You can find the dird in /scratch1/NCEPDEV/stmp2/Terry.McGuinness/RUNDIRS/C48_ATM_5a4e4de6/postsnd.282649/out_gfs_bufr_120

Please check the setting of DIRD. I suspect that the length of fnbufr is more than 150 and that caused the job to fail.

**Comment by WalterKolczynski-NOAA** [comment]
@KateFriedman-NOAA Try NOAA-EMC/gfs-utils#22. If that works, we can merge that and then update this PR.

**Comment by KateFriedman-NOAA** [comment]
> Please check the setting of DIRD. I suspect that the length of fnbufr is more than 150 and that caused the job to fail.

@BoCui-NOAA Thanks for taking a look and finding the likely cause of the failure!

> Try https://github.com/NOAA-EMC/gfs-utils/pull/22. If that works, we can merge that and then update this PR.

@WalterKolczynski-NOAA Will do, thanks! Stay tuned...

**Comment by KateFriedman-NOAA** [comment]
@WalterKolczynski-NOAA @BoCui-NOAA The length modification to the gfs_bufr code made by Walter in https://github.com/NOAA-EMC/gfs-utils/pull/22 appears to have worked. I copied the failed CI test to my own space on Hera, cloned the updated gfs-utils into my global-workflow clone, and reran the failed gfspostsnd job. It completed without failure. See the log and output here on Hera:
```
PR code: /scratch1/NCEPDEV/global/Kate.Friedman/git/pr1834/sorc/gfs_utils.fd
log: /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/C48_ATM_5a4e4de6/logs/2021032312/gfspostsnd.log
output: /scratch1/NCEPDEV/stmp4/Kate.Friedman/comrot/C48_ATM_5a4e4de6/gfs.20210323/12/products/atmos/bufr/
```

**Comment by WalterKolczynski-NOAA** [comment]
The gfs-utils PR has been merged. Update the gfs-utils version to [a283262](https://github.com/NOAA-EMC/gfs-utils/commit/a283262b26efeb63c550e927768fb0d00869f5a1).

File: parm/config/gfs/config.base.emc.dyn
Comment (WalterKolczynski-NOAA) [inline parm/config/gfs/config.base.emc.dyn]: Did you actually want to change the default behavior to make bufr soundings? Or is this an accidental commit?

**Review by WalterKolczynski-NOAA — DISMISSED** [review]
and the gfs-utils update

File: parm/config/gfs/config.base.emc.dyn
Comment (KateFriedman-NOAA) [reply parm/config/gfs/config.base.emc.dyn]: Changed this temporarily for the CI tests so the postsnd job is included in the gfs suites. Will change back before merging the PR.

**Comment by KateFriedman-NOAA** [comment]
Updated gfs-utils hash to [a283262](https://github.com/NOAA-EMC/gfs-utils/commit/a283262b26efeb63c550e927768fb0d00869f5a1). Will rerun CI tests to confirm all is truly well now.

File: parm/config/gfs/config.base.emc.dyn
Comment (WalterKolczynski-NOAA) [reply parm/config/gfs/config.base.emc.dyn]: After submitting, it dawned on me that was the likely reason. Just leave this as unresolved until you change back.

**Comment by KateFriedman-NOAA** [comment]
Rerun CI tests on Hera and Orion have passed. Rerunning test on WCOSS2 to double check updated gfs_bufr code is all good.

File: parm/config/gfs/config.base.emc.dyn
Comment (KateFriedman-NOAA) [reply parm/config/gfs/config.base.emc.dyn]: I have reverted it back to "NO" at https://github.com/NOAA-EMC/global-workflow/pull/1834/commits/139a5499e1f61400bd00d1a5cb97225c692af180. Will now resolve this conversation and re-request your review.

**Comment by BoCui-NOAA** [comment]
> Rerun CI tests on Hera and Orion have passed. Rerunning test on WCOSS2 to double check updated gfs_bufr code is all good.

@KateFriedman-NOAA Can you tell me the location of the re-run results on WCOSS2? Thanks!

**Comment by KateFriedman-NOAA** [comment]
> @KateFriedman-NOAA Can you tell me the location of the re-run results on WCOSS2? Thanks!

@BoCui-NOAA see the following on Dogwood, it just finished running the gfspostsnd job:

expdir: /lfs/h2/emc/global/noscrub/kate.friedman/expdir/test96
output/logs: /lfs/h2/emc/ptmp/kate.friedman/comrot/test96
clone: /lfs/h2/emc/global/noscrub/kate.friedman/git/develop/workflow

**Comment by BoCui-NOAA** [comment]
> > @KateFriedman-NOAA Can you tell me the location of the re-run results on WCOSS2? Thanks!
> 
> @BoCui-NOAA see the following on Dogwood, it just finished running the gfspostsnd job:
> 
> expdir: /lfs/h2/emc/global/noscrub/kate.friedman/expdir/test96 output/logs: /lfs/h2/emc/ptmp/kate.friedman/comrot/test96 clone: /lfs/h2/emc/global/noscrub/kate.friedman/git/develop/workflow

@KateFriedman-NOAA 
/lfs/h2/emc/ptmp/kate.friedman/comrot/test96/logs/2021122100/gfspostsnd.log

SENDDBN=NO in the above log file. Is it set only for the CI test?  There is no files copied to /lfs/h2/emc/ptmp/kate.friedman/comrot/test96/gfs.20211221/00//products/atmos/wmo if SENDDBN=NO. 

All the others look good to me. Thanks!

**Comment by KateFriedman-NOAA** [comment]
> SENDDBN=NO in the above log file. Is it set only for the CI test? There is no files copied to /lfs/h2/emc/ptmp/kate.friedman/comrot/test96/gfs.20211221/00//products/atmos/wmo if SENDDBN=NO.

@BoCui-NOAA thanks for checking and mentioning this! `SENDDBN` was still set to its default value of "NO" in my test. I have set it to "YES" and rerun the job on WCOSS2. Please see the output in the `wmo` folder now:

```
kate.friedman@dlogin06:/lfs/h2/emc/ptmp/kate.friedman/comrot/test96> ll gfs.20211221/00/products/atmos/wmo/
total 52936
-rw-r--r-- 1 kate.friedman emc  8273734 Sep  7 15:52 gfs_collective1.postsnd_00
-rw-r--r-- 1 kate.friedman emc 10871630 Sep  7 15:52 gfs_collective2.postsnd_00
-rw-r--r-- 1 kate.friedman emc  6099408 Sep  7 15:52 gfs_collective3.postsnd_00
-rw-r--r-- 1 kate.friedman emc  8527876 Sep  7 15:52 gfs_collective4.postsnd_00
-rw-r--r-- 1 kate.friedman emc  6410026 Sep  7 15:52 gfs_collective5.postsnd_00
-rw-r--r-- 1 kate.friedman emc  4433366 Sep  7 15:52 gfs_collective6.postsnd_00
-rw-r--r-- 1 kate.friedman emc  1637804 Sep  7 15:51 gfs_collective7.postsnd_00
-rw-r--r-- 1 kate.friedman emc  4150986 Sep  7 15:52 gfs_collective8.postsnd_00
-rw-r--r-- 1 kate.friedman emc  3783892 Sep  7 15:52 gfs_collective9.postsnd_00

```

Also, here is the updated log: /lfs/h2/emc/ptmp/kate.friedman/comrot/test96/logs/2021122100/gfspostsnd.log

Do you recommend that when `DO_BUFRSND="YES"` is set (and the job is included in the suite) that `SENDDBN="YES"` is also automatically set?

> All the others look good to me. Thanks!

Excellent, thanks!

**Comment by BoCui-NOAA** [comment]
> kate.friedman@dlogin06:/lfs/h2/emc/ptmp/kate.friedman/comrot/test96> ll gfs.20211221/00/products/atmos/wmo/

The files in directory wmo look good. Thanks!
 
> Do you recommend that when `DO_BUFRSND="YES"` is set (and the job is included in the suite) that `SENDDBN="YES"` is also automatically set?

Will `SENDDBN="YES"` affect other parts of the whole gfs testing? If not, it is ok for me to set `DO_BUFRSND="YES" and then `SENDDBN="YES". 

Thanks!

**Comment by KateFriedman-NOAA** [comment]
> The files in directory wmo look good. Thanks!

Great, thanks for checking and confirming!

> Will `SENDDBN="YES"` affect other parts of the whole gfs testing?

Good point...other jobs do use that variable. I will leave `SENDDBN="NO"` by default for now and reevaluate later for the dev mode setting (it will be "YES" for ops).

**Comment by KateFriedman-NOAA** [comment]
Test on WCOSS2 was successful. This PR is ready for final approval and merging. @WalterKolczynski-NOAA please review/approve when able, thanks!

**Review by aerorahul — APPROVED** [review]
looks good.

**Pull request merged**
